x-pack/plugins/task_manager/server/lib/task_partitioner.ts

```diff
@@ -38,9 +43,16 @@ export class TaskPartitioner {
   }
 
   async getPartitions(): Promise<number[]> {
-    const allPodNames = await this.getAllPodNames();
-    const podPartitions = assignPodPartitions(this.podName, allPodNames, this.allPartitions);
-    return podPartitions;
+    const lastUpdated = new Date(this.podPartitionsLastUpdated).getTime();
+    const now = Date.now();
+
+    // update the pod partitions cache after 10 seconds
+    if (now - lastUpdated >= CACHE_INTERVAL) {
+      const allPodNames = await this.getAllPodNames();
```

Comment:
If there's an error retrieving this, should we return the cached partitions?

Reply:
Yeah that sounds good, resolved in this commit [e39827c](https://github.com/elastic/kibana/pull/189562/commits/e39827c923269686eef997b89d1917e8586fd8c4)